python text_mining.py

```diff
@@ -0,0 +1,117 @@
+import requests
+import re, math
+from collections import Counter
+from operator import itemgetter
+import operator
+
+def text_preprocessing(text):
+    """Check if the the input is a string or pathfile. convert it to string
+
+    >>> text_preprocessing(4)
+
+    >>> text_preprocessing('4')
+    '4'
+    """
+    if isinstance(text, str):
+        if text.endswith('.txt'):
+            path = text
+            text = open(path,'r')
+            return text.read()
+        else:
+            return text
+    else:
+        return None
+
+def word_counter(text):
+    """return a list with word frequencies
+
+    >>>
+    """
+    WORD = re.compile(r'\w+')
```

Comment:
I would caution against using all caps for variable names because some programs will interpret them differently.